src/EnergyPlus/SolarShading.cc

```diff
@@ -3867,6 +4267,21 @@ namespace SolarShading {
         NVTEMP = 0;
         KK = 0;
 
+        //Check if clipping polygon is rectangle
+        if (SlaterandBarsky) {
+            auto l1(HCA.index(NS2, 1));
+            bool rectFlag = ((NV2 == 4) && (((((HCX[l1] == HCX[l1 + 1] && HCY[l1] != HCY[l1 + 1]) &&
+                                               ((HCY[l1 + 2] == HCY[l1 + 1] && HCY[l1 + 3] == HCY[l1]))) &&
+                                              HCX[l1 + 2] == HCX[l1 + 3]) ||
+                                             ((((HCY[l1] == HCY[l1 + 1] && HCX[l1] != HCX[l1 + 1]) &&
+                                                (HCX[l1 + 2] == HCX[l1 + 1] && HCX[l1 + 3] == HCX[l1])) &&
+                                               (HCY[l1 + 2] == HCY[l1 + 3]))))));
```

Comment:
As an FYI, native C++ operators will short circuit, so if NV2 is not 4, it will not evaluate the rest of this.  But that said, yeah, this is a big messy expression.  I would create three individual terms:

```c
bool const numVertsIs4 = NV2 == 4;
bool const term1 = blah;
bool const term2 = blah;
rectFlat = numVertsIs4 || term1 || term2;
```

Of course better naming but that would at least make the logic here tractable.